src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3269,6 +3293,17 @@ internal bool CheckRefEscape(SyntaxNode node, BoundExpression expr, uint escapeF
                         escapeTo,
                         checkingReceiver: false,
                         diagnostics);
+
+                case BoundKind.Conversion:
+                    var conversion = (BoundConversion)expr;
+                    if (conversion.Conversion == Conversion.ImplicitThrow)
+                    {
+                        return CheckRefEscape(node, conversion.Operand, escapeFrom, escapeTo, checkingReceiver, diagnostics);
```

Comment:
We are bound to hit `case BoundKind.ThrowExpression` through this, right? We just call into it in this way for consistency?

Reply:
Yes, this is just for consistency.